astropy/wcs/wcsapi/utils.py

```diff
@@ -60,13 +65,25 @@ def wcs_info_str(wcs):
             'Bounds\n')
     # fmt: on
 
+    if wcs.pixel_bounds is None:
+        pixel_bounds = [None for _ in range(wcs.pixel_n_dim)]
+    else:
+        # converting to scalar arrays and back to Python with np.array(val).item()
+        # guarantees that we end up with Python scalars (int or float) with
+        # simple reprs, while not making any unnecessary type promotion
+        # (e.g. int to float)
+        pixel_bounds = [
+            tuple(np.array(b).item() for b in bounds) for bounds in wcs.pixel_bounds
+        ]
+
     for ipix in range(wcs.pixel_n_dim):
         # fmt: off
         s += (('{0:' + str(pixel_dim_width) + 'g}').format(ipix) + '  ' +
                 ('{0:' + str(pixel_nam_width) + 's}').format(wcs.pixel_axis_names[ipix] or 'None') + '  ' +
                 (" " * 5 + str(None) if pixel_shape[ipix] is None else
                 ('{0:' + str(pixel_siz_width) + 'g}').format(pixel_shape[ipix])) + '  ' +
-                '{:s}'.format(str(None if wcs.pixel_bounds is None else wcs.pixel_bounds[ipix]) + '\n'))
```

Comment:
Was this one actually a problem? Since it was passed in to `str`, I would think the representation would not change. That would also remove the need for the less elegant of the overrides...

Reply:
Yes, it is a problem because in the application that @Cadair showcased, `wcs.pixel_bounds[ipix]` is a tuple (of numpy scalars). My test actually covers this, so you may convince yourself by checking out the test without the patch :)

Reply:
It's weird because `str(numpy_int)` and `str(int)` should still be the same (sorry, have to run, can check later).

Reply:
Ah, I didn't realize `wcs.pixel_bounds[ipix]` itself returned a tuple. Should have looked better...